### PR TITLE
[WIP] DUOS-1606 Filter Archived Collections from Queries

### PR DESCRIPTION
**Pull request by quazi-h**
[DUOS-1606](https://broadworkbench.atlassian.net/browse/DUOS-1606)
Updates to SQL query logic for DarCollectionDAO and DataAccessRequestDAO calls.
Added a functional test for each DAO query that was changed.
Could probably improve upon the setup methods used between the tests since there's a lot of repetition.

----
Have you read [CONTRIBUTING.md](../CONTRIBUTING.md) lately? If not, do that first.

- Label PR with a Jira ticket number and include a link to the ticket
- Label PR with a security risk modifier [no, low, medium, high]
- PR describes scope of changes
- Get a minimum of one thumbs worth of review, preferably two if enough team members are available
- Get PO sign-off for all non-trivial UI or workflow changes
- Verify all tests go green
- Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: src/main/java/org/broadinstitute/consent/http/db/DarCollectionDAO.java [hunk 1]
```diff
@@ -50,9 +46,11 @@ public interface DarCollectionDAO {
       DarCollection.FILTER_TERMS_QUERY +
       " )";
 
+  String archiveFilterQuery = " AND ((dar.data #>> '{}')::jsonb->>'status' != 'Archived' OR (dar.data #>> '{}')::jsonb->>'status' IS NULL) ";
```
Comment (rushtong) [inline src/main/java/org/broadinstitute/consent/http/db/DarCollectionDAO.java]: Can we make the clause case insensitive? Here, and throughout:
```suggestion
  String archiveFilterQuery = " AND (LOWER(data->>'status') != 'archived' OR (dar.data #>> '{}')::jsonb->>'status' IS NULL) ";
```

File: src/main/java/org/broadinstitute/consent/http/db/DataAccessRequestDAO.java [hunk 2]
```diff
@@ -32,7 +32,8 @@ public interface DataAccessRequestDAO extends Transactional<DataAccessRequestDAO
   @SqlQuery(
       "SELECT id, reference_id, collection_id, parent_id, draft, user_id, create_date, sort_date, submission_date, update_date, (data #>> '{}')::jsonb AS data FROM data_access_request "
           + "  WHERE not (data #>> '{}')::jsonb ??| array['partial_dar_code', 'partialDarCode'] "
-          + "  AND draft != true ")
+          + "  AND draft != true "
+          + "  AND (data->>'status' != 'Archived' OR data->>'status' IS NULL ) ")
```
Comment (rushtong) [inline src/main/java/org/broadinstitute/consent/http/db/DataAccessRequestDAO.java]: Same thing in this file
```suggestion
          + "  AND (LOWER(data->>'status') != 'archived' OR data->>'status' IS NULL ) ")
```

**Review by rushtong — CHANGES_REQUESTED** [review]
One small request, then 👍🏽

**Review by JVThomas — COMMENTED** [review]
Seconding Greg's comments. The queries look right to me, they just need to do a case insensitive check on status

**Comment by quazi-h** [comment]
I can't get my tests to run - whether I try running the whole file of DAOTests or an individual one - there is an error in the class configuration and all the tests get ignored. Here's the message I see on my console when it fails:
`Test ignored.

java.lang.RuntimeException: java.io.IOException: Failed to bind to 0.0.0.0/0.0.0.0:8180

	at org.eclipse.jetty.setuid.SetUIDListener.lifeCycleStarting(SetUIDListener.java:229)
	at org.eclipse.jetty.util.component.AbstractLifeCycle.setStarting(AbstractLifeCycle.java:205)
	at org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:72)
	at io.dropwizard.cli.ServerCommand.run(ServerCommand.java:53)
	at io.dropwizard.cli.EnvironmentCommand.run(EnvironmentCommand.java:60)
	at io.dropwizard.cli.ConfiguredCommand.run(ConfiguredCommand.java:98)
	at io.dropwizard.testing.DropwizardTestSupport.startIfRequired(DropwizardTestSupport.java:326)
	at io.dropwizard.testing.DropwizardTestSupport.before(DropwizardTestSupport.java:225)
	**at org.broadinstitute.consent.http.db.DAOTestHelper.startUp(DAOTestHelper.java:112)**
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.RunBefores.invokeMethod(RunBefores.java:33)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:24)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:69)
	at com.intellij.rt.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:33)
	at com.intellij.rt.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:235)
	at com.intellij.rt.junit.JUnitStarter.main(JUnitStarter.java:54)
Caused by: java.io.IOException: Failed to bind to 0.0.0.0/0.0.0.0:8180
	at org.eclipse.jetty.server.ServerConnector.openAcceptChannel(ServerConnector.java:349)
	at org.eclipse.jetty.server.ServerConnector.open(ServerConnector.java:310)
	at org.eclipse.jetty.setuid.SetUIDListener.lifeCycleStarting(SetUIDListener.java:216)
	... 25 more
Caused by: java.net.BindException: Address already in use
	at java.base/sun.nio.ch.Net.bind0(Native Method)
	at java.base/sun.nio.ch.Net.bind(Net.java:555)
	at java.base/sun.nio.ch.ServerSocketChannelImpl.netBind(ServerSocketChannelImpl.java:337)
	at java.base/sun.nio.ch.ServerSocketChannelImpl.bind(ServerSocketChannelImpl.java:294)
	at java.base/sun.nio.ch.ServerSocketAdaptor.bind(ServerSocketAdaptor.java:89)
	at org.eclipse.jetty.server.ServerConnector.openAcceptChannel(ServerConnector.java:344)
	... 27 more`
	
	DAOTestHelper.java:112
	`testApp.before();`
	
	I'm a little confused as I didn't change anything in DAOTestHelper and my tests were running earlier this week when I was developing.

**Comment by rushtong** [comment]
@quazi-broad DAO tests are running fine for me - and it looks like they're running fine in github actions as well. Try closing out all running docker+non-docker apps and then running tests:
```
mvn clean test
```
Bind issues are usually a conflict issue, i.e. something else is running on that same port at the same time. From the above message, it might be the UI. If you're just running tests in one place, like the IDE or in a terminal window, you should be fine.

**Review by rushtong — APPROVED** [review]
These look good, thank you for the changes!

**Review by JVThomas — APPROVED** [review]

**Pull request merged**